sound/hda/hdac_stream.c

```diff
@@ -423,7 +433,7 @@ int snd_hdac_stream_setup_periods(struct hdac_stream *azx_dev)
 	azx_dev->frags = 0;
 
 	pos_adj = bus->bdl_pos_adj;
-	if (!azx_dev->no_period_wakeup && pos_adj > 0) {
+	if (runtime && !azx_dev->no_period_wakeup && pos_adj > 0) {
```

Comment:
could we have this change on line 436 as a separate patch, I don't fully get what is being tested here, and if it's related to the compressed support

Reply:
Separated "preparation changes" in v6 with commit:
ALSA: hda: Prepare for compress stream support

Reply:
As code within the scope of that 'if' is performing position adjustments, opted-out of it for compress streams (runtime can be not-NULL only in PCM case) as I didn't find any value in that. Format is ignored by probes, really.